Tests/MaakuTests/Core/DocumentConverterSpec.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  DocumentConverterSpec.swift
+//  Maaku
+//
+//  Created by Tim Learmont on 5/25/19.
+//  Copyright © 2019 Kristopher Baker. All rights reserved.
+//
+
+import Maaku
+import Nimble
+import Quick
+import XCTest
+
+// swiftlint:disable line_length
+class DocumentConverterSpec: QuickSpec {
+    enum ParseError: LocalizedError {
+        case customError(message: String)
+        var localizedDescription: String {
+            switch self {
+            case .customError(let message):
+                return message
+            }
+        }
+    }
+
+    static func checkHeading(item: Block, identifier: String, expectedLevel: HeadingLevel, expectedText: String) throws {
+        guard let heading = item as? Heading else {
+            // It wasn't a heading. Report
+            throw ParseError.customError(message: "\(identifier): wasn't a heading, it was \(item)")
+        }
+        if heading.level != expectedLevel {
+            throw ParseError.customError(message: "\(identifier): expected heading level \(expectedLevel), but got \(heading.level)")
+        }
+        // We only expect 1 item for a heading
+        if heading.items.count != 1 {
+            throw ParseError.customError(message: "\(identifier): expected heading to only have 1 item, but it had \(heading.items)")
+        }
+        try DocumentConverterSpec.checkText(item: heading.items[0], identifier: identifier, expectedText: expectedText)
+    }
+
+    static func checkText(item: Inline, identifier: String, expectedText: String) throws {
+        guard let text = item as? Text else {
+            throw ParseError.customError(message: "\(identifier): expected Text, got \(item)")
+        }
+        if text.text != expectedText {
+            throw ParseError.customError(message: "\(identifier): expected \(expectedText), but got \(text.text)")
+        }
+    }
+
+    static func checkParagraph(item: Block, identifier: String, expectedText: [String?]) throws {
+        guard let paragraph = item as? Paragraph else {
+            throw ParseError.customError(message: "\(identifier): expected Paragraph, got \(item)")
+        }
+        if paragraph.items.count != expectedText.count {
+            throw ParseError.customError(message: "\(identifier): expected \(expectedText.count) items, but got \(paragraph.items.count)")
+        }
+        // swiftlint:disable identifier_name
+        for i in 0 ..< paragraph.items.count {
+            if let expectedString = expectedText[i] {
+                try DocumentConverterSpec.checkText(item: paragraph.items[i], identifier: "\(identifier) [item \(i)]", expectedText: expectedString)
+            } else {
+                // If the value in the expectedText array is nil, that means we expect a softbreak
+                if let item = paragraph.items[i] as? SoftBreak {
```

Comment:
Xcode is showing a warning here since `item` is unused. Could this be changed to 

```swift
if paragraph.items[i] is SoftBreak {
```

or even change the entire if-else to a guard, like 

```swift
guard paragraph.items[i] is SoftBreak else {
    throw ParseError.customError(message: "\(identifier): expected item \(i) to be a soft break, but got \(paragraph.items[i])")
}
```